packages/salesforcedx-utils/src/cli/checkCliVersion.ts

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2023, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { execSync } from 'child_process';
+import * as semver from 'semver';
+
+export enum CheckCliEnum {
+  validCli = 1,
+  outdatedSFDXVersion = 2,
+  onlySFv1 = 3,
+  cliNotInstalled = 4,
+  bothSFDXAndSFInstalled = 5
+}
+
+export class CheckCliVersion {
```

Comment:
Consider renaming this class w/o a verb in the name, such as CliVersion. Class names tend to have noun based names. Same would apply to the enum.